### ISSUE TITLE
suboptimal_flops: can't call method `mul_add` on ambiguous numeric type `{float}

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#10015]
### Summary

.

### Lint Name

suboptimal_flops

### Reproducer

I tried this code:

```rust
pub fn main() {
    let pi = 3.1415927;
    println!("{:?}", -pi * (pi + 2.0 / pi) - pi * 5.0);
}
```

I saw this happen:

```
This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0689]: can't call method `mul_add` on ambiguous numeric type `{float}`
 --> src/main.rs:3:28
  |
3 |     println!("{:?}", (-pi).mul_add(pi + 2.0 / pi, -pi * 5.0));
  |                            ^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0689`.
Original diagnostics will follow.

warning: multiply and add expressions can be calculated more efficiently and accurately
 --> src/main.rs:3:22
  |
3 |     println!("{:?}", -pi * (pi + 2.0 / pi) - pi * 5.0);
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using: `(-pi).mul_add(pi + 2.0 / pi, -pi * 5.0)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#suboptimal_flops
  = note: requested on the command line with `-W clippy::suboptimal-flops`
```

### Version

```text
rustc 1.67.0-nightly (c97b539e4 2022-11-30)
binary: rustc
commit-hash: c97b539e408ea353f4fde2f9251d598291fec421
commit-date: 2022-11-30
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
